### ISSUE TITLE
IndexFactory only uses GenericKeys

### ISSUE THREAD

**Issue by apavlo** [cmu-db/peloton#431]
The IntsKey stuff is commented out. I will fix.

**Comment by apavlo** [comment]
All of the PRIMARY KEY indexes are getting stored with the same name:

```
[DEBUG] [src/index/index_factory.cpp:277:GetInstance()] 2016-12-24 08:42:58 - Index 'customer_pkey' => BWTREE::IntsKey<2>(s_w_id, s_i_id)
[DEBUG] [src/index/index_factory.cpp:277:GetInstance()] 2016-12-24 08:42:58 - Index 'customer_pkey' => BWTREE::IntsKey<2>(d_w_id, d_id)
[DEBUG] [src/index/index_factory.cpp:277:GetInstance()] 2016-12-24 08:42:58 - Index 'customer_pkey' => BWTREE::IntsKey<3>(no_w_id, no_d_id, no_o_id)
[DEBUG] [src/index/index_factory.cpp:277:GetInstance()] 2016-12-24 08:42:58 - Index 'customer_pkey' => BWTREE::IntsKey<2>(s_w_id, s_i_id)
```

**Comment by wangziqi2013** [comment]
Is this fixed?

**Comment by apavlo** [comment]
Yes, this is fixed in #432
